### ISSUE TITLE
Wrong parameters for some German health insurance cards on Android

### ISSUE THREAD

**Issue by dfyx** [FeitianSmartcardReader/bR301_iOS_SDK_RELEASE#4]
I'm trying to read some German health insurance cards and most of them work but for some older test cards (ATR `3B 9F 18 81 B1 FE 45 1F 03 00 64 04 10 10 08 31 C0 73 F6 11 C0 07 90 00 4A`)  I get either `DK.TRANS_RETURN_ERROR` or `DK.CARD_TIMEOUT` when calling `transApdu`.

After a lot of digging and directly accessing `CardCmd` and `ProtocolExtended` I found out that there's a problem with `Catr.analysis_atr`. When I skip that call, those cards work fine but in exchange some newer cards (ATR `3B D0 97 FF 81 B1 FE 45 1F C7 EB`) don't.

**In short:** `Catr.analysis_atr` sets wrong parameters when parsing ATR `3B 9F 18 81 B1 FE 45 1F 03 00 64 04 10 10 08 31 C0 73 F6 11 C0 07 90 00 4A`.

**Comment by dfyx** [comment]
Okay, I played with it a little more and it looks like your translation from `Di` to `D` is wrong. You define the following table:

```Java
static int[] Di = { -1, 1, 2, 4, 8, 16, -1, -1, 512, 768, 1024, 1536, 2048, -1, -1 };
```

The correct values according to [Wikipedia](https://en.wikipedia.org/wiki/Answer_to_reset#Interface_byte_TA1) are:

```Java
static int[] Di = { -1, 1, 2, 4, 8, 16, 32, 64, 12, 20, -1, -1, -1, -1, -1, -1 };
```

**Comment by dfyx** [comment]
There's also a related bug when calculating the baud rate. You check if the calculated rate is greater than 150 and set it to that value if it is. The problem is that the value you are comparing is a signed byte which interprets 150 as -106. Therefore almost all realistic values are replaced with -106 which doesn't work for some cards.

**Comment by dfyx** [comment]
After fixing those two bugs with a Java Bytecode Editor (is the source code for the android library available somewhere?) all of my test cards work.

**Comment by FeitianSmartcardReader** [comment]
@dfyx We re-write android SDK and please refer below:
The new android SDK provide C by PCSCLite API and jni with private API.

Any questions, you can comment on below repo, thanks
https://github.com/FeitianSmartcardReader/FEITIAN_MOBILE_READERS/tree/master/Android%20SDK
